babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -987,3 +987,51 @@ SqlNode AlternativeTypeConversionQuery(SqlNode query) :
         return SqlStdOperatorTable.CAST.createCall(s.end(this), args);
     }
 }
+
+SqlNode RankSortingExpression() :
+{
+    SqlNode e;
+}
+{
+    e = Expression(ExprContext.ACCEPT_SUB_QUERY)
+    (
+        <ASC>
+    |
+        [ <DESC> ]  {
+            e = SqlStdOperatorTable.DESC.createCall(getPos(), e);
+        }
+    )
+    { return e; }
+}
+
+// Parse RANK function call with sorting expressions.
+SqlNode RankFunctionCallWithParams() :
+{
+    final SqlFunctionCategory funcType = SqlFunctionCategory.USER_DEFINED_FUNCTION;
+    final SqlNodeList orderByList = new SqlNodeList(SqlParserPos.ZERO);
+    final SqlIdentifier qualifiedName;
+    final SqlNode rankCall;
+    final SqlNode over;
+    SqlNode e;
+    Span s;
```

Comment:
Would it be reasonable to have the variable "s" and "s1" to be final?

Reply:
Yeah those should probably be final if they're not being reassigned